plugins/importexport/csv/CSVImportExportPlugin.inc.php

```diff
@@ -80,192 +197,724 @@ function executeCLI($scriptName, &$args) {
+		}
+	}
+
+	/**
+	 * Display the command-line usage information
+	 */
+	function usage($scriptName) {
+		echo __('plugins.importexport.csv.cliUsage', [
+			'scriptName' => $scriptName,
+			'pluginName' => $this->getName()
+		]) . "\n";
+	}
+
+	/**
+	 * Parse and validate initial command args
+	 *
+	 * @param string $scriptName
+	 * @param array $args
+	 * @return string[]
+	 */
+	private function parseCommandLineArguments($scriptName, $args) {
 		$filename = array_shift($args);
 		$username = array_shift($args);
+		$basePath = dirname($filename);
 
 		if (!$filename || !$username) {
 			$this->usage($scriptName);
-			exit();
+			exit(1);
+		}
+
+		return [$filename, $username, $basePath];
+	}
+
+	/**
+	 * Retrieve and validate the User by username
+	 *
+	 * @param string $username
+	 */
+	private function validateUser($username) {
+		$user = $this->getUser($username);
+		if (!$user) {
+			echo __('plugins.importexport.csv.unknownUser', ['username' => $username]) . "\n";
+			exit(1);
+		}
+
+		$this->_userId = $user->getId();
+	}
+
+	/**
+	 * @param string $filename
+	 * @param 'r'|'w' $mode
+	 */
+	private function createNewFile($filename, $mode) {
+		try {
+			return new SplFileObject($filename, $mode);
+		} catch (Exception $e) {
+			echo $e->getMessage();
+			exit(1);
+		}
+	}
+
+	/**
+	 * @param string $filename
+	 */
+	private function createAndValidateCSVFile($filename) {
+		$file = $this->createNewFile($filename, 'r');
+		$file->setFlags(SplFileObject::READ_CSV);
+
+		$headers = $file->fgetcsv();
+
+		$missingHeaders = array_diff($this->_expectedHeaders, $headers);
+
+		if (count($missingHeaders)) {
+			echo __('plugin.importexport.csv.missingHeadersOnCsv', ['missingHeaders' => $missingHeaders]);
+			exit(1);
+		}
+
+		return $file;
+	}
+
+	/**
+	 * @param string $basePath
+	 */
+	private function createAndValidateCSVFileInvalidRows($csvForInvalidRowsName) {
+		$invalidRowsFile = $this->createNewFile($csvForInvalidRowsName, 'w');
+		$invalidRowsFile->fputcsv(array_merge(array_pad($this->_expectedHeaders, $this->_expectedRowSize, null), ['error']));
+
+		return $invalidRowsFile;
+	}
+
+	/**
+	 * Insert static data that will be used for the submission processing
+	 */
+	private function initializeStaticVariables() {
+		$this->_dirNames = Application::getFileDirectories();
+		$this->_format = trim($this->_dirNames['context'], '/') . '/%d/' . trim($this->_dirNames['submission'], '/') . '/%d';
+
+		$this->_fileManager = new FileManager();
+		$this->_publicFileManager = new PublicFileManager();
+
+		$this->_fileService = Services::get('file');
+		$this->_publicationService = Services::get('publication');
+	}
+
+	/**
+	 * Returns a cached DAO or create a new one, if it isn't initialized yet.
+	 *
+	 * @param string $daoType
+	 */
+	private function getCachedDao($daoType) {
+		return $this->_daos[$daoType] ?? $this->_daos[$daoType] = DAORegistry::getDAO($daoType);
+	}
```

Comment:
It's not needed to cache these `DAO`s, they are cheap to get. But as it's already implemented, I don't think it's worth to revert. So, this is just a note :)